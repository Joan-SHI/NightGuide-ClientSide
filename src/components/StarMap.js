import React from 'react';
import {
    StyleSheet,
    WebView,
    View
} from 'react-native';

export default class StarMap extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {

        return (
            <View style={styles.container}>
                <WebView style={styles.webView}
                    renderError={() => errorMessage}
                    scalesPageToFit={true}
                    source={{ uri: 'https://virtualsky.lco.global/embed/?longitude=-119.86286000000001&latitude=34.4326&projection=polar&constellationlabels=true' }}
                    allowTransparency="true"
                />
            </View>
        );
    };
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'stretch'
    },
    webView: {
        marginBottom: -355,
        marginTop: 0,
        backgroundColor: 'black'
    }
});