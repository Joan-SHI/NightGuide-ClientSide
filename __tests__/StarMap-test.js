import React from 'react';
import StarMap from '../src/components/StarMap';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<StarMap />).toJSON();
  expect(tree).toMatchSnapshot();
});