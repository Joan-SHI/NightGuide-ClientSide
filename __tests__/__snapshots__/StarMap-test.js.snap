// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTWebView
      messagingEnabled={false}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      scalesPageToFit={true}
      source={
        Object {
          "uri": "https://virtualsky.lco.global/embed/?longitude=-119.86286000000001&latitude=34.4326&projection=polar&constellationlabels=true",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "backgroundColor": "black",
            "marginBottom": -500,
          },
        ]
      }
    />
  </View>
</View>
`;
